src/api/model/body_model.py

```diff
@@ -49,12 +50,50 @@ def __init__(self):
         self.graph = nx.from_pandas_edgelist(result, 'SOURCE_SUBREDDIT', 'TARGET_SUBREDDIT', create_using=nx.DiGraph())
 
 
-    def get_sentiments(self, target):
-        posts_for_target = self.data.loc[self.data['TARGET_SUBREDDIT'] == target]
-        posts_for_target = posts_for_target.sort_values(by=['DATE','TIMEOFDAY'])
-        sentiments = list(posts_for_target['LINK_SENTIMENT'])
-        return sentiments 
+    def get_top_target_subreddits(self, num):
+        return self.data.groupby(["TARGET_SUBREDDIT"]).size().reset_index(name="counts") \
+            .sort_values("counts", ascending=False).head(num)
+
+    def get_sentiments(self, target_subreddit, source_subreddit):
         
+        FIRST_DATE_IN_DATA_SET = '01-01-2014'
+        LATEST_DATE_IN_DATA_SET = '12-31-2017'
+        daterange = pd.date_range(FIRST_DATE_IN_DATA_SET, LATEST_DATE_IN_DATA_SET).astype(str)
+
+        self.sentimentboxhelper = SentimentBoxHelper(self.data)
+        self.MIN_SENTIMENT, self.MAX_SENTIMENT = self.sentimentboxhelper.run()
```

Comment:
This should not be here actually, I tried to make it compute `MIN_SENTIMENT` and `MAX_SENTIMENT` by placing this code snippet in `__init__`, but it would not work

Reply:
Create the sentboxhelper in the `__init__`, so you do not have to recreate it each time the API call is made, since it is based on the constant `self.data` anyway. Seems like `MIN_SENTIMENT` and `MAX_SENTIMENT` are not used?